firestore/integration_test_internal/src/query_snapshot_test.cc

```diff
@@ -49,94 +48,139 @@ TEST_F(QuerySnapshotTest, Assignment) {
 
 #endif  // defined(__ANDROID__)
 
-TEST_F(QuerySnapshotTest, Equality) {
+TEST_F(QuerySnapshotTest, Equality1) {
   CollectionReference collection =
       Collection({{"a", {{"k", FieldValue::String("a")}}},
                   {"b", {{"k", FieldValue::String("b")}}},
                   {"c", {{"k", FieldValue::String("c")}}}});
   QuerySnapshot snapshot1 = ReadDocuments(collection.Limit(2));
   QuerySnapshot snapshot2 = ReadDocuments(collection.Limit(2));
-  QuerySnapshot snapshot3 = ReadDocuments(collection.Limit(1));
-  QuerySnapshot snapshot4 = ReadDocuments(collection);
-  QuerySnapshot snapshot5 =
+
+  EXPECT_TRUE(snapshot1 == snapshot1);
```

Comment:
Why are you changing the QuerySnapshot tests in this PR for aggregate queries? It seems out of scope. If these changes really have nothing to do with aggregate queries, please move these changes to a new PR against the main branch and I'll happily review. I'm all for improving test coverage, but I have an aversion to mashing unrelated changes into a PR.